gui/wxpython/gis_set.py

```diff
@@ -510,9 +512,13 @@ def SuggestDatabase(self):
         if self.GetRCValue("LOCATION_NAME") != "<UNKNOWN>":
             return
         path = get_possible_database_path()
-        # If nothing found, try to create GRASS directory
+
+        # If nothing found, try to create GRASS directory and copy startup loc
         if path is None:
             path = create_database_directory()
+            location = get_startup_location()
+            if location:
+                copy_startup_location(path, location)
```

Comment:
You are ignoring the return value from `copy_startup_location()`. Perhaps, you can wrap this whole thing into one function, maybe named `get_startup_location()`, which will return the location/mapset on success and None otherwise and internally resolve all the cases (no location to copy, copying failed, and, possibly, also an existing user's copy of the location).

Perhaps more importantly, it is not clear to me if this is meant to actually start in a mapset in this location. Creating this location if there is no other location is good, but actually using it, means picking also a mapset and starting in that one. With that, the overarching function should be something like `get_mapset_in_startup_location()` and it would take care of all above plus picking, probably creating, a mapset to use.

Reply:
Well, actually, if you want to leave this PR just at creating the location and solving the actual starting in some mapset there in another PR, that's fine with me.

Reply:
It starts now for the first-time user.